### PR TITLE
Blocked error messages: towards an ultimate solution.

### PR DESCRIPTION
**Pull request by dbrant**
This uses some exotic API manipulation to get the wiki-specific Blocked error page, and populate it with the correct parsed parameters (no longer requiring us to use a local string resource).
The only downside of this method is that we need to use three (!) API requests to get all the necessary information. Perhaps this can be streamlined in a future update.

### CODE REVIEW

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsTasksFragment.kt [hunk 1]
```diff
@@ -204,7 +208,7 @@ class SuggestedEditsTasksFragment : Fragment() {
                 }
                 .observeOn(AndroidSchedulers.mainThread())
                 .doAfterTerminate {
-                    if (blockInfo != null) {
+                    if (blockMessage != null) {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsTasksFragment.kt:211]: Would it be better if we use `!blockMessage.isNullOrEmpty()`?

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsTasksFragment.kt [hunk 2]
```diff
@@ -213,7 +217,7 @@ class SuggestedEditsTasksFragment : Fragment() {
                         isPausedOrDisabled = true
                     }
 
-                    if (!isPausedOrDisabled && blockInfo == null) {
+                    if (!isPausedOrDisabled && blockMessage == null) {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsTasksFragment.kt:220]: `blockMessage.isNullOrEmpty()`?

File: app/src/main/java/org/wikipedia/util/ThrowableUtil.kt [hunk 3]
```diff
@@ -91,26 +94,30 @@ object ThrowableUtil {
...
-        val app = WikipediaApp.getInstance()
-        var retStr = "<b>${app.getString(R.string.error_blocked_title)}</b>"
-        val title = PageTitle(UserAliasData.valueFor(app.appOrSystemLanguageCode), info.blockedBy, app.wikiSite)
+    @WorkerThread
+    fun getBlockMessageHtml(blockInfo: MwServiceError.BlockInfo): String {
+        var html = ""
+        Observable.zip(ServiceFactory.get(WikipediaApp.getInstance().wikiSite).userInfo,
+            ServiceFactory.get(WikipediaApp.getInstance().wikiSite).parsePage("MediaWiki:Blockedtext"),
+            ServiceFactory.get(WikipediaApp.getInstance().wikiSite).parseText(blockInfo.blockReason),
+            { userInfoResponse: MwQueryResponse, blockedParseResponse: MwParseResponse, reasonParseResponse: MwParseResponse ->
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/util/ThrowableUtil.kt:103]: Explicit type specifications can be removed.

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsTasksFragment.kt (hunk 1 above)
Comment (dbrant) [reply app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsTasksFragment.kt]: correct!

File: app/src/main/java/org/wikipedia/util/ThrowableUtil.kt (hunk 3 above)
Comment (dbrant) [reply app/src/main/java/org/wikipedia/util/ThrowableUtil.kt]: done!

**Pull request merged**